src/main/java/org/broadinstitute/hellbender/utils/help/HelpConstants.java

```diff
@@ -8,4 +8,22 @@ public static String forumPost(String post) {
         return GATK_FORUM_URL + post;
     }
 
+    /**
+     * Definition of the group names / categories of tools.
+     * The names get parsed to make supercategories in the doc index,
+     * so be careful when making big changes -- see GATKDoclet.java toMap()
```

Comment:
Although there is some redundancy, I don't think we can completely unify them, since they don't completely intersect. There are documented features that are not CommandLinePrograms (like ReadFilters, and perhaps others such as transformers, annotations, etc), as well as vice versa. In order to eliminate much of the redundancy, the doc system uses the DocumentedFeature annotation as a tag interface to tag a class as a documentation target. When it creates the doc, it prefers to use the DocumentedFeature annotation values, but if they're not present, it will fall back to the  CommandLineProgramGroup ones if they're present. So most CommandLinePrograms just need to have an empty DocumentedFeature tag annotation to enable the doc. ReadFilters would want a fully populated DocumentedFeature annotation, since there will be no fallback.